### ISSUE TITLE
accessibilityLabel for Text not being read out on some android devices

### ISSUE THREAD

**Issue by jsiServiceNSW** [facebook/react-native#30180]
## Description
When assigning an accessibilityLabel for Text, it is not read out by our accessibility reader on our Pixel 3. It seems to work fine on samsung devices. 

## React Native version:
Running 0.62.2

## To reproduce
1. add the following line `<Text accessibilityLabel={'apple'}>Hi there</Text>`
2. On a Pixel 3 use an accessibility scanner and use text to speech to read out the Text component. 

## What occurs
Reads out the child of the text component
In the above example reads out 'Hi there'

## Expected Results
Read out accessibility label value instead of text value in child.
In the above example reads out 'apple'

## Known devices that have the issue
- Google Pixel 3 (Android 10 and Android 11)

## Known devices that don't have the issue
- Samsung galaxy S9
- Huawei P20
- Oppo A57

**Comment by jsiServiceNSW** [comment]
After investigating this further the issue was actually the screen reader that we were using. TalkBack works perfectly fine and will read the accessibility label. However, Select to Speak does not read out the accessibility label and will simply read out the text.
